Desktop_Interface/uartstyledecoder.h

```diff
@@ -34,17 +34,27 @@ class uartStyleDecoder : public QObject
     uint32_t dataBit_max = 7;
     unsigned short currentUartSymbol = 0;
     bool jitterCompensationNeeded = true;
+
     void updateSerialPtr(double baudRate, unsigned char current_bit);
-    unsigned char getNextUartBit();
+    unsigned char getUartBit(int bitIndex) const;
     void decodeNextUartBit(unsigned char bitValue);
```

Comment:
I'd recommend keeping this as a state machine too.  That's what makes the most sense, given the FIFOness of it.